src/com/google/javascript/jscomp/Compiler.java

```diff
@@ -668,6 +668,8 @@ private void compileInternal() {
     if (!precheck()) {
       return;
     }
+    
+    alwaysPreChecks(); // even in whitespace only mode
 
     if (!options.skipNonTranspilationPasses || options.lowerFromEs6()) {
       check();
```

Comment:
We actually already have a case where whitespace only mode runs the checks (when we are transpiling from ES6 down to ES3/5), so it's reasonable to just run check unconditionally, and add this logic to the beginning of getChecks in DefaultPassConfig.